binderhub/static/js/pages/LoadingPage.jsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect, useState } from "react";
+import { BuilderLauncher } from "../components/BuilderLauncher.jsx";
+import { useParams } from "react-router";
+import { useSearchParams } from "react-router-dom";
+import { NBViewerIFrame } from "../components/NBViewerIFrame.jsx";
+import { LoadingIndicator } from "../components/LoadingIndicator.jsx";
+import { FaviconUpdater } from "../components/FaviconUpdater.jsx";
+import { LaunchSpec, Spec } from "../spec.js";
+
+/**
+ * @typedef {object} LoadingPageProps
+ * @prop {URL} baseUrl
+ * @param {LoadingPageProps} props
+ * @returns
+ */
+export function LoadingPage({ baseUrl }) {
+  const [progressState, setProgressState] = useState(null);
+
+  const params = useParams();
+  const buildSpec = params["*"];
+
+  const [searchParams, _] = useSearchParams();
+
+  const [isLaunching, setIsLaunching] = useState(false);
+
+  const spec = new Spec(buildSpec, LaunchSpec.fromSearchParams(searchParams));
+
+  useEffect(() => {
+    // Start launching after the DOM has fully loaded
+    setTimeout(() => setIsLaunching(true), 1);
+  }, []);
```

Comment:
This looks a bit flaky. Is there a specific element you're waiting for in the DOM tree? If so how is it created?

Reply:
I think I was trying to trigger this useEffect (https://github.com/jupyterhub/binderhub/pull/1856/files#diff-17ed39387f14113cf305cbfd7cc95014fc635d5988acbe542676dd87695c967cR185) and couldn't figure out how else to do it :( In general I think I was trying to trigger 'actions' in one component based on events in a different component, and without using reducers i think i'm hackily using these 'is' booleans to do so